wallet/src/de/schildbach/wallet/service/CoinJoinService.kt

```diff
@@ -115,38 +127,31 @@ class CoinJoinMixingService @Inject constructor(
     private var mixingCompleteListeners: ArrayList<MixingCompleteListener> = arrayListOf()
     private var sessionCompleteListeners: ArrayList<SessionCompleteListener> = arrayListOf()
 
-    override var mode: CoinJoinMode = CoinJoinMode.BASIC
+    var mode: CoinJoinMode = CoinJoinMode.NONE
     override var mixingStatus: MixingStatus = MixingStatus.NOT_STARTED
         private set
 
     private val coroutineScope = CoroutineScope(
-        Executors.newFixedThreadPool(2).asCoroutineDispatcher(),
+        Executors.newFixedThreadPool(2, ContextPropagatingThreadFactory("coinjoin-pool")).asCoroutineDispatcher()
     )
 
+    private val uiCoroutineScope = CoroutineScope(Dispatchers.Main)
+
     private var blockChain: AbstractBlockChain? = null
     private val isBlockChainSet: Boolean
         get() = blockChain != null
     private var networkStatus: NetworkStatus = NetworkStatus.UNKNOWN
+    private var hasAnonymizableBalance: Boolean = false
 
     // https://stackoverflow.com/questions/55421710/how-to-suspend-kotlin-coroutine-until-notified
-    private val mixingMutex = Mutex(locked = true)
+//    private val mixingMutex = Mutex(locked = true)
     private val updateMutex = Mutex(locked = false)
+    private val updateMixingStateMutex = Mutex(locked = false)
     private var exception: Throwable? = null
-    override suspend fun waitForMixing() {
-        log.info("Waiting for mixing to complete by watching lock...")
-        mixingMutex.withLock {}
-        log.info("Mixing complete according to released lock")
-    }
-    override suspend fun waitForMixingWithException() {
-        waitForMixing()
-        exception?.let {
-            log.error("Mixing error: {}", it.message, it)
-            throw it
-        }
-    }
-    private fun setMixingComplete() {
-        mixingMutex.unlock()
-    }
```

Comment:
Since mixing is not part of another process, there is no need for these waiting functions.